File: sdk/typescript/src/intelligence.ts
/**
 * TypeScript SDK Client for Intelligence Platform
 */

export class IntelligenceClient {
  private baseUrl: string;
  private apiKey: string;

  constructor(baseUrl: string, apiKey: string = "") {
    this.baseUrl = baseUrl;
    this.apiKey = apiKey;
  }

  async ingestSignal(source: string, signalType: string, message: str, tenantId: string = "global"): Promise<any> {
    const res = await fetch(`${this.baseUrl}/v1/intelligence/signals`, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ source, signal_type: signalType, message, tenant_id: tenantId }),
    });
    return res.json();
  }

  async runAnalysis(tenantId: string = "global", resourceId: string = "res_1"): Promise<any> {
    const res = await fetch(`${this.baseUrl}/v1/intelligence/analyze?tenant_id=${tenantId}&resource_id=${resourceId}`, {
      method: "POST",
    });
    return res.json();
  }

  async listInsights(tenantId: string = "global"): Promise<any[]> {
    const res = await fetch(`${this.baseUrl}/v1/intelligence/insights?tenant_id=${tenantId}`);
    return res.json();
  }

  async listRecommendations(tenantId: string = "global"): Promise<any[]> {
    const res = await fetch(`${this.baseUrl}/v1/intelligence/recommendations?tenant_id=${tenantId}`);
    return res.json();
  }
}

File: sdk/go/intelligence.go
package llmengine

import (
	"context"
	"fmt"
)

type IntelligenceClient struct {
	baseURL string
	apiKey  string
}

func NewIntelligenceClient(baseURL string, apiKey string) *IntelligenceClient {
	return &IntelligenceClient{baseURL: baseURL, apiKey: apiKey}
}

func (c *IntelligenceClient) IngestSignal(ctx context.Context, source string, signalType string, message string, tenantID string) (string, error) {
	return fmt.Sprintf("IngestSignal %s/%s for tenant %s from %s", source, signalType, tenantID, c.baseURL), nil
}

func (c *IntelligenceClient) ListRecommendations(ctx context.Context, tenantID string) (string, error) {
	return fmt.Sprintf("ListRecommendations for tenant %s from %s", tenantID, c.baseURL), nil
}
